components/ILIAS/Forum/classes/Tables/ForumStatisticsTable.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\UI\Component\Table\Data as DataTable;
+use ILIAS\UI\Factory as UIFactory;
+use Psr\Http\Message\ServerRequestInterface as HttpRequest;
+use ILIAS\UI\Component\Table\DataRetrieval;
+use ILIAS\UI\Renderer as UIRenderer;
+use ILIAS\Data\Order;
+use ILIAS\Data\Range;
+
+class ForumStatisticsTable implements DataRetrieval
+{
+    private UIFactory $ui_factory;
+    private ilLanguage $lng;
+    private bool $has_active_lp = false;
+    /** @var int[] */
+    private array $completed = [];
+    /** @var int[] */
+    private array $failed = [];
+    /** @var int[] */
+    private array $in_progress = [];
+    private HttpRequest $request;
+    private UIRenderer $ui_renderer;
+    private ?array $records = null;
+
+    public function __construct(
+        private readonly ilObjForum $forum,
+        private readonly ilForumProperties $obj_properties,
+        private readonly bool $has_general_lp_access,
+        private readonly bool $has_rbac_or_position_access,
+        private readonly ilObjUser $actor,
+    ) {
+        global $DIC;
```

Comment:
Can we remove the `$DIC` usage an try to pass all dependencies via `__construct`?